Add Spanish translations

PavDev3 opened the pull request:
Added Spanish translations

pangolp commented:
The file is missing the extension, it should be called: `esES.lua`. However, I don't know if that is enough for the translations to be displayed in Spanish, I think it is necessary to modify another file. In my case, I also modified the following file. `RXPGuides/locale/locales.xml`
```diff
<Ui xmlns="http://www.blizzard.com/wow/ui/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.blizzard.com/wow/ui/..\FrameXML\UI.xsd">
    <Script file="enUS.lua"/>
+   <Script file="esES.lua"/>
    <Script file="frFR.lua"/>
    <Script file="zhCN.lua"/>
    <Script file="zhTW.lua"/>
    <Script file="deDE.lua"/>
    <Script file="koKR.lua"/>
    <Script file="NPCnames.lua"/>
</Ui>

```

PavDev3 commented:
> The file is missing the extension, it should be called: `esES.lua`. However, I don't know if that is enough for the translations to be displayed in Spanish, I think it is necessary to modify another file. In my case, I also modified the following file. `RXPGuides/locale/locales.xml`
> 
> ```diff
> <Ui xmlns="http://www.blizzard.com/wow/ui/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.blizzard.com/wow/ui/..\FrameXML\UI.xsd">
>     <Script file="enUS.lua"/>
> +   <Script file="esES.lua"/>
>     <Script file="frFR.lua"/>
>     <Script file="zhCN.lua"/>
>     <Script file="zhTW.lua"/>
>     <Script file="deDE.lua"/>
>     <Script file="koKR.lua"/>
>     <Script file="NPCnames.lua"/>
> </Ui>
> ```

Thank for you comment

I updated and fix a few files.

PavDev3 commented:
> This does not translate the texts of the guides, but it helps at least to start making the translations into Spanish. For the guides, it will be a little more difficult, because the format of the same is a single text, and it is all in English. Therefore, the text will have to be concatenated a little, and in that way, be able to make the translations... but it will be a complex job...

I've checked other translation PRs and I don't see any more files. In my case, I don't have the game to check right now, but I was hoping that with this update, the addons would work fine.

If anyone can tell me the correct process to do it perfectly, I'd appreciate it.

pangolp commented:
> > This does not translate the texts of the guides, but it helps at least to start making the translations into Spanish. For the guides, it will be a little more difficult, because the format of the same is a single text, and it is all in English. Therefore, the text will have to be concatenated a little, and in that way, be able to make the translations... but it will be a complex job...
> 
> I've checked other translation PRs and I don't see any more files. In my case, I don't have the game to check right now, but I was hoping that with this update, the addons would work fine.
> 
> If anyone can tell me the correct process to do it perfectly, I'd appreciate it.

It's complicated, because the code is all in one text. We'll have to find a way to concatenate or work in a different way. With this, some translations are applied, but not the guide itself, the guide will always remain in English. Check out this file for example, so you have an example and understand what I'm talking about...

https://github.com/RestedXP/RXPGuides/blob/main/Guides/SurvivalGuide/A-Hardcore-18-19%20Loch%20Modan.lua

If you look at it, you will see that the text is in English, regardless of the client used. It is not just a problem with Spanish, but with all languages. In all languages, the guide is always shown in English, at least in Classic, and changing this is not going to be easy... although it is not impossible either, but several things will have to be reviewed, possibly.

PavDev3 commented:
Ohhh, I see! That’s true. We could use AI to translate all of them, but yes, maybe we can approach it differently because is too much

SabreValkyrn commented:
> If you look at it, you will see that the text is in English, regardless of the client used. It is not just a problem with Spanish, but with all languages. In all languages, the guide is always shown in English, at least in Classic, and changing this is not going to be easy... although it is not impossible either, but several things will have to be reviewed, possibly.

That's what the [getForeign function tries to address](https://github.com/RestedXP/RXPGuides/blob/cc9f78ae63b7222d333fb25648e14db229e6feaf/Locale.lua#L16). Translates known words when possible, and at least UI elements use in-game global variables, e.g. `_G.ENABLE`, but that doesn't mean much for the guides itself.

The problem is there are hundreds of premium guides, not just these embedded guides. So all translation needs done on the addon/client-side, which is very taxing for the WoW client/addon.

SabreValkyrn commented:
Looks good, thank you!

Will be live in the next release.
